config/runtime.exs

```diff
@@ -1,5 +1,40 @@
 import Config
 
+defmodule ConfigParser do
+  def parse_external_ip(ip) do
+    with {:ok, parsed_ip} <- ip |> to_charlist() |> :inet.parse_address() do
+      parsed_ip
+    else
+      _ ->
+        raise("""
+        Bad EXTERNAL_IP format. Expected IPv4, got: \
+        #{inspect(ip)}
+        """)
+    end
+  end
+
+  def parse_port_range(range) do
+    with [str1, str2] <- String.split(range, "-"),
+         from when from in 0..65_535 <- String.to_integer(str1),
+         to when to in from..65_535 and from <= to <- String.to_integer(str2) do
+      {from, to}
+    else
+      _else ->
+        raise("""
+        Bad PORT_RANGE enviroment variable value. Expected "from-to", where `from` and `to` \
```

Comment:
As we are talking about a specific PORT_RANGE environment variable in the error's message, I would assume that `parse_port_range` would read that variable by itself - which in fact does not happen, since we are passing a result of `System.get_env("PORT_RANGE", "50000-59999")` invocation there. One could use `parse_port_range` to parse env with any other name and the error would still say: "Bad PORT_RANGE ...". Perhaps `parse_port_range` could accept the env name and the default value as it's arguments? (The similar comment is applicable also to the `parse_external_ip` function, and in case you have decided to follow my advice I would consider renaming the `ConfigParser` module :D )

Reply:
I changed it slightly, take a look now.